brave/brave-core
# Support sending f1 to typed f4 addresses

Resolves https://github.com/brave/brave-browser/issues/30401
Support sending f1->f4.
1) FilAddress is extended to support new f4 namespace. f4 address consists of 2 identifiers - agent id and address.
We currently support only agent id "10".
2) fvm_shared lib is updated to 3.3.0 to support new f4 addresses to sign.
2) When f4 address is selected as recipient "Method" field should be changed  to 3844450837 for EstimateGas and MPoolPush operations.

Update:
fvm_shared lib is forked to brave-org repo (brave-experiments until PR is approved).
Library was stripped to depend on old versions: https://github.com/brave-experiments/ref-fvm/pull/2

Update:
Upstream of the fvm_shared lib is used

Sec review: 
https://github.com/brave/security/issues/1290



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
1) Create Filecoin account, obtain some funds
2) Take default Ethereum account and translate it address to f410 address using https://beryx.zondax.ch/address_converter
3) Use obtained f410 address when send from Filecoin account
4) Check that transaction succeeds, blockexplorer correctly shows info, account has the transaction in the history
